forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java

```diff
@@ -0,0 +1,174 @@
+             * the ShowPost fragment, so that the user can access the forum contents
+             *
+             * @param username The username of the user that successfully logged in
+             * @author Alberto Valente (g2), Taulant Bullaku (g2)
+             */
+            @Override
+            public void onLoginSuccess(String username) {
+
+                if(username != null) {
+                    //creates the username bundle
+                    Bundle bundle = new Bundle();
+                    bundle.putString(KEY_USERNAME, username);
+
+                    //creates an instance of the fragment to be launched
+                    ShowPosts showPostsFragment = new ShowPosts();
+                    //passes the bundle to the fragment as an argument
+                    showPostsFragment.setArguments(bundle);
+
+                    //performs the fragment transaction
+                    getActivity()
+                            .getSupportFragmentManager()
+                            .beginTransaction()
+                            .replace(
+                                    R.id.fragmentContainer,
+                                    showPostsFragment
+                            )
+                            .commit();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_USERNAME);
+                }
+            }
+
+            /**
+             * The method of the listener is triggered when the database response to
+             * the login request is negative and the view model requests to show the
+             * user an error message, that is performed through the use of a toast
+             *
+             * @param message The error message about what was wrong with the login request
+             * @author Alberto Valente (g2), Taulant Bullaku (g2)
+             */
+            @Override
+            public void onLoginFailure(String message) {
+
+                if (message != null) {
+                    Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_ERROR_MESSAGE);
+                }
+            }
+        });
+
+        //When the login button is clicked a request is sent through the view model
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                //The username and password inserted are gathered from the EditText objects
+                userName = usernameEditText.getText().toString();
+                userPwd = pwdEditText.getText().toString();
+
+                //Then the credentials are handed to the view model method to be checked
+                viewModel.loginToForum(getContext(), userName, userPwd);
+            }
+        });
+
+    }
+
+    public interface OnFragmentInteractionListener
```

Comment:
I see that you initialized "mListener" at line 36 but it does not do anything. Does that TODO mean it's a work in progress or it just specifies what the method onFragmentInteraction should do? I also noticed there are three public interfaces with this same name on different classes of the code you sent me. Why is that? Could it be a good idea to write on another file and implement the interface in the classes? Maybe there's something i'm missing though.